homeassistant/core.py

```diff
@@ -1189,7 +1189,6 @@ def async_set(
         This method must be run in the event loop.
         """
         entity_id = entity_id.lower()
-        new_state = str(new_state)
```

Comment:
I think we should keep this as an extra guard. Potentially log an error if it's not a string but still cope.